stdlib/http/src/main/ballerina/src/http/auth/basic_auth_handler.bal

```diff
@@ -49,14 +49,20 @@ public type BasicAuthHandler object {
     # Authenticates the incoming request with the use of the credentials passed as the Basic Auth header.
     #
     # + req - The request object.
-    # + return - Returns `true` if it is possible to authenticate with Basic Auth. Else, returns `false` or the `error` in case of an error.
-    public function process(Request req) returns boolean|error {
+    # + return - Returns `true` if it is possible to authenticate with Basic Auth. Else, returns `false` or
+    # the `AuthenticationError` in case of an error.
```

Comment:
Shall we use `http:AuthenticationError` here ?

Reply:
Yes. I have missed that. Thanks for pointing.

Reply:
We'll address all the documentation of the error types in a different PR.